iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/VoltageRegulationSerDe.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.serde.extensions;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedSet;
+import com.powsybl.commons.extensions.ExtensionSerDe;
+import com.powsybl.commons.io.DeserializerContext;
+import com.powsybl.commons.io.SerializerContext;
+import com.powsybl.iidm.network.Injection;
+import com.powsybl.iidm.network.ReactiveLimitsHolder;
+import com.powsybl.iidm.network.extensions.VoltageRegulation;
+import com.powsybl.iidm.network.extensions.VoltageRegulationAdder;
+import com.powsybl.iidm.serde.IidmVersion;
+import com.powsybl.iidm.serde.NetworkDeserializerContext;
+import com.powsybl.iidm.serde.NetworkSerializerContext;
+import com.powsybl.iidm.serde.TerminalRefSerDe;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Coline Piloquet {@literal <coline.piloquet@rte-france.fr>}
+ */
+@AutoService(ExtensionSerDe.class)
+public class VoltageRegulationSerDe<T extends Injection<T> & ReactiveLimitsHolder> extends AbstractVersionableNetworkExtensionSerDe<T, VoltageRegulation<T>> {
+
+    public VoltageRegulationSerDe() {
+        super("voltageRegulation", VoltageRegulation.class, "vr",
```

Comment:
I don't know if we want it just for IIDM 1.12/ if we can add it for the other versions. Also do we want in core the second version of the extension for IIDM 1.0?

Reply:
@geofjamg help :-)